include/SongEditor.h

```diff
@@ -45,15 +46,44 @@ class Song;
 class TextFloat;
 class TimeLineWidget;
 
+Q_DECLARE_METATYPE(QLinearGradient)
+
 class positionLine : public QWidget
 {
+	Q_OBJECT
+	Q_PROPERTY ( bool tailGradient READ tailGradient WRITE setTailGradient )
+	Q_PROPERTY ( QColor lineColor READ lineColor WRITE setLineColor )
 public:
-	positionLine( QWidget * parent );
+	positionLine ( QWidget * parent, ComboBoxModel * zoom, Song * song );
+	
+	int width ();
+	void go ( int x, int y );	// NOTE: Check SongEditor.cpp for usage
+	void zoomUpdate ();
+	void playStateChanged ();
+	
+	// qproperty access functions
+	bool tailGradient () const;
+	void setTailGradient ( const bool & g );
```

Comment:
Didn't exactly get what you mean by this...?

Reply:
The `&` in `const bool & g` means that rather than passing the boolean value directly, a reference to a value stored elsewhere is passed. Typically references are used if the original object passed to the function needs to modified, or if the object is expensive to copy. Neither of these are the case here, so `bool g` is sufficient for the parameter.

(Minor nitpick - perhaps `hasTailGradient` and `setHasTailGradient` would be better names? The current ones sound like they access the gradient itself, rather than whether it is present or not.)

Reply:
Well, I wrote that function based off of the other functions for reading qproprties, and all of them had the & symbol there. Maybe it’s something QT requires? Either ways, I’d rather be on the safe side here especially when the rest of the code uses references for bool.

About the tail gradient rename, I’ll commit with the modified name soon.

Reply:
Most of the other `Q_PROPERTY` setters in LMMS work with `QColor`, which comes under "expensive to copy". It's not actually that expensive, but the rule of thumb is generally "has a user-defined copy constructor, or is larger than a pointer", both of which `QColor` satisfies. Other `Q_PROPERTY(bool)`s in LMMS don't use references, for example:
https://github.com/LMMS/lmms/blob/5d7e6720e1ba0c154041fd663801f7a7a0e16cb5/include/PianoRoll.h#L73
https://github.com/LMMS/lmms/blob/5d7e6720e1ba0c154041fd663801f7a7a0e16cb5/include/PianoRoll.h#L156

Reply:
Okay then, I'll make it a `bool`.